app/src/main/java/com/android/ootd/ui/post/FitCheckViewModel.kt

```diff
@@ -2,24 +2,45 @@ package com.android.ootd.ui.post
 
 import android.net.Uri
 import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.android.ootd.model.items.ItemsRepository
+import com.android.ootd.model.items.ItemsRepositoryProvider
 import kotlinx.coroutines.flow.MutableStateFlow
 import kotlinx.coroutines.flow.StateFlow
 import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.launch
 
 data class FitCheckUIState(
     val image: Uri = Uri.EMPTY,
     val description: String = "",
     val errorMessage: String? = null,
     val invalidPhotoMsg: String? = null,
+    val isLoading: Boolean = false,
+    val successMessage: String? = null,
+    val savedPostId: String? = null
 ) {
   val isPhotoValid: Boolean
     get() = invalidPhotoMsg == null && image != Uri.EMPTY
 }
 
-class FitCheckViewModel : ViewModel() {
+class FitCheckViewModel(
+    private val repository: ItemsRepository = ItemsRepositoryProvider.repository
+) : ViewModel() {
   private val _uiState = MutableStateFlow(FitCheckUIState())
   val uiState: StateFlow<FitCheckUIState> = _uiState.asStateFlow()
 
+  fun deleteItemsForPost(postUuid: String) {
+    viewModelScope.launch {
+      try {
+        val itemsRepo = ItemsRepositoryProvider.repository
+        itemsRepo.deletePostItems(postUuid)
+      } catch (e: Exception) {
+        // optional: log or show error
+        e.printStackTrace()
```

Comment:
Maybe instead of printing the whole stack you could just print a log error:

```suggestion
        Log.e("FitCheckViewModel", "Failed to delete items for post: $postUuid", e)
```

I think it makes more sense and follow what we did in the rest of the project, lmkwyt ?